tests/nested/manual/uc-update-assets-secure-add-sbat/task.yaml

```diff
@@ -0,0 +1,162 @@
+    export QUILT_REFRESH_ARGS="-p ab --no-timestamps --no-index"
+    export QUILT_PATCHES=debian/patches
+    quilt push -a
+    quilt new werror.patch
+    quilt add Make.defaults
+    sed -i "s/-Werror //" Make.defaults
+    quilt refresh
+    git add debian/patches/werror.patch
+    apt build-dep -y ./
+    dpkg-buildpackage -us -uc
+    cp debian/shim/usr/lib/shim/shimx64.efi ../old-shimx64.efi
+    ./debian/rules clean
+    rm -rf .pc
+  )
+  # Download and build shim version with sbat support
+  (
+    cd shim
+    git checkout -f debian/15.4-0ubuntu9-0-g9a8d2b1b2965eee962893f94d3ae6d08f2e4cf94
+    gbp export-orig
+    apt build-dep -y ./
+    dpkg-buildpackage -us -uc
+    cp debian/shim-*/*/shimx64.efi ../new-shimx64.efi
+    ./debian/rules clean
+    rm -rf .pc
+  )
+
+  CHANNEL="$(tests.nested show version)/stable"
+  snap download --basename=pc --channel="${CHANNEL}" pc
+  unsquashfs -d pc pc.snap
+  KEY_NAME=$(tests.nested download snakeoil-key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  # Remove signatures on grub
+  while sbverify --list pc/grubx64.efi | grep "^signature [0-9]*$"; do
+    sbattach --remove pc/grubx64.efi
+  done
+  # Remove sbat on grub if it did exist
+  objcopy --remove-section=.sbat pc/grubx64.efi || true
+  # Save that file
+  cp pc/grubx64.efi grubx64.efi
+
+  cp old-shimx64.efi pc/shim.efi.signed
+
+  # Repack pc gadget for the initial image
+  tests.nested secboot-sign file pc/shim.efi.signed "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  tests.nested secboot-sign file pc/grubx64.efi "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  old_shim_sha="$(sha256sum pc/shim.efi.signed | sed "s/ .*//")"
+  old_grub_sha="$(sha256sum pc/grubx64.efi | sed "s/ .*//")"
+  snap pack pc "$(tests.nested get extra-snaps-path)"
+
+  tests.nested build-image core
+  tests.nested create-vm core
+
+  cat <<EOF >sbat.csv
+  sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md
+  grub,1,Free Software Foundation,grub,2.06,https://www.gnu.org/software/grub/
+  grub.ubuntu,1,Canonical,grub2,2.06-2ubuntu7,https://packages.ubuntu.com/source/jammy/grub2
+  EOF
+  sbatsize="$(($(stat -c %b sbat.csv)*$(stat -c %B sbat.csv)))"
+  truncate --size "${sbatsize}" sbat.csv
+
+  # Add .sbat at the end
+  grubsize="$(($(stat -c %b grubx64.efi)*$(stat -c %B grubx64.efi)))"
+  objcopy --add-section .sbat=sbat.csv --change-section-address .sbat="${grubsize}" grubx64.efi grubx64-sbat.efi
+
+  cp new-shimx64.efi pc/shim.efi.signed
+  cp grubx64-sbat.efi pc/grubx64.efi
+
+  # Sign modified grub and shim
+  tests.nested secboot-sign file pc/shim.efi.signed "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  tests.nested secboot-sign file pc/grubx64.efi "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  new_shim_sha="$(sha256sum pc/shim.efi.signed | sed "s/ .*//")"
+  new_grub_sha="$(sha256sum pc/grubx64.efi | sed "s/ .*//")"
+
+  if [ "${UPDATE_SEED}" = true ]; then
+    (cd pc && python3 "${TESTSLIB}/bump_asset_edition.py" --role system-seed)
```

Comment:
I now noticed that I was doing the same in my fde branch test, but using `yq` command, for instance:
`yq -i '(.volumes.pc.structure | with_entries(select(.value.name == "ubuntu-boot")) | .[].update.edition) |= . + 1' "$gadget_p"`
Not that I have a preference, as `yq` syntax is terrible and difficult to understand, although it can be used more generically to modify any part in the yaml.

Reply:
Do you install `yq`? Maybe it is faster not to. For the tests what is fastest is probably best.

Reply:
I do install it, yes. That is done also in another test. In fact I see also lots of places were we look at or change bits of yaml files using tools like sed or yaml2json and so on that are probably best served by standardizing on `yq`. We should install it by default in fact to minimize re-installs in the VMs.

Reply:
I have pushed a commit to use yq. I will squash it if the tests pass.

Note that I have simplified your expression to:

```
(.volumes.pc.structure[] | select(.role == "system-seed") | .update.edition) |= . + 1
```

Reply:
@alfonsosanchezbeato Have a look at the last commit. Then I will squash it if it is fine.

Reply:
LGTM, thank you

Reply:
Thank you. I squashed it in the previous commit.